### PR TITLE
Some more balance changes for the dev build

### PR DESCRIPTION
**Pull request by Grobobobo**

### CODE REVIEW

File: LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Effect_BattlefieldAwarenessCooldown.uc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+//---------------------------------------------------------------------------------------
+//  FILE:    X2Effect_BattlefieldAwareness
+//  AUTHOR:  Grobobobo
+//  PURPOSE: A dummy effect That represents the cooldown of battlefield awareness
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Effect_BattlefieldAwarenessCooldown.uc:4]: Do you even need this? Can't you just use `X2Effect_Persistent` and set its `EffectName` when you attach it to the ability?

File: LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Effect_BloodTrail_LW.uc [hunk 2]
```diff
@@ -46,11 +46,8 @@ private function bool ShouldApplyBonuses(XComGameState_Effect EffectState, XComG
...
 
-	if (AbilityState.SourceWeapon == EffectState.ApplyEffectParameters.ItemStateObjectRef)
-	{
-		Target.GetUnitValue('DamageThisTurn', DamageUnitValue);
-		return DamageUnitValue.fValue > 0;
-	}
+	Target.GetUnitValue('DamageThisTurn', DamageUnitValue);
+	return DamageUnitValue.fValue > 0;
 
 	return false;
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Effect_BloodTrail_LW.uc:52]: I'm surprised this build with 2 return statements.
```suggestion
```

File: LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2LWModTemplate_SkirmisherAbilities.uc [hunk 3]
```diff
@@ -47,6 +50,9 @@ static function UpdateAbilities(X2AbilityTemplate Template, int Difficulty)
 	case 'TotalCombat':
 		UpdateTotalCombat(Template);
 		break;
+	case'CombatPresence':
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2LWModTemplate_SkirmisherAbilities.uc:53]:
```suggestion
	case 'CombatPresence':
```

**Review by pledbrook — CHANGES_REQUESTED** [review]
Looks pretty solid. And hardly any formatting changes :wink: Anyway, there are a few other issues and requests I've added inline. Perhaps remove Battlemaster from Skirmisher and add Grenade pocket to Total Combat.

File: LongWarOfTheChosen/Localization/XComGame.int [hunk 4]
```diff
@@ -5352,11 +5354,13 @@ LocPromotionPopupText="<Bullet/> Reduces Shadow's cooldown by 1 turn.<br/>"
...
 LocFriendlyName="Blood Trail"
-LocLongDescription="Shots deal +<Ability:BLOODTRAILDAMAGE/> damage and reduce target's Dodge by -<Ability:BLOOD_TRAIL_ANTIDODGE_BONUS/> if the target has been wounded this turn."
-LocHelpText="Shots deal +<Ability:BLOODTRAILDAMAGE/> damage and reduce target's Dodge by -<Ability:BLOOD_TRAIL_ANTIDODGE_BONUS/> if the target has been wounded this turn."
-LocPromotionPopupText="<Bullet/> Only applies on the same turn the enemy is damaged.<br/><Bullet/> Plan ahead with squadmate attacks to take full advantage of Blood Trail.<br/><Bullet/> Only works with the primary weapon.<br/>"
+LocLongDescription="Attacks deal +<Ability:BLOODTRAILDAMAGE/> damage and reduce target's Dodge by -<Ability:BLOOD_TRAIL_ANTIDODGE_BONUS/> if the target has been wounded this turn."
+LocHelpText="Attacks deal +<Ability:BLOODTRAILDAMAGE/> damage and reduce target's Dodge by -<Ability:BLOOD_TRAIL_ANTIDODGE_BONUS/> if the target has been wounded this turn."
+LocPromotionPopupText="<Bullet/> Only applies on the same turn the enemy is damaged.<br/><Bullet/> Plan ahead with squadmate attacks to take full advantage of Blood Trail.<br/><Bullet/> Works with all weapons.<br/>"
```
[the remark this answers is not in the document]
Comment (Grobobobo) [reply LongWarOfTheChosen/Localization/XComGame.int:5362]: It currently does, though i wonder if it should? No harm in it working, right?

File: LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Effect_BattlefieldAwarenessCooldown.uc (hunk 1 above)
Comment (Grobobobo) [reply LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Effect_BattlefieldAwarenessCooldown.uc:4]: That was the OG Idea, but I've created the effect when it didn't work to check if some things didn't work. Not sure if I should change back at this point.

File: LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Ability_SkirmisherAbilitySet_LW.uc [hunk 5]
```diff
@@ -359,3 +368,155 @@ static function X2AbilityTemplate Interrupt_LW()
...
+	Template.AbilitySourceName = 'eAbilitySource_Perk';
+	Template.eAbilityIconBehaviorHUD = eAbilityIconBehavior_NeverShow;
+	Template.Hostility = eHostility_Neutral;
+
+	Template.AbilityToHitCalc = default.DeadEye;
+	Template.AbilityTargetStyle = default.SelfTarget;
+	
+	EventListenerTrigger = new class'X2AbilityTrigger_EventListener';
+	EventListenerTrigger.ListenerData.Deferral = ELD_OnStateSubmitted;
+	EventListenerTrigger.ListenerData.EventID = 'UnitDied';
```
[the remark this answers is not in the document]
Comment (Grobobobo) [reply LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Ability_SkirmisherAbilitySet_LW.uc:397]: I was not 100% sure how killmail differed from UnitDied and decided to not risk it and just get it done. Is that a problem?

File: LongWarOfTheChosen/Localization/XComGame.int (hunk 4 above)
[the remark this answers is not in the document]
Comment (pledbrook) [reply LongWarOfTheChosen/Localization/XComGame.int]: I feel strongly either way. I just care about clarity for the player.

File: LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Ability_SkirmisherAbilitySet_LW.uc (hunk 5 above)
[the remark this answers is not in the document]
Comment (pledbrook) [reply LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Ability_SkirmisherAbilitySet_LW.uc:397]: They differ in when they occur ('KillMail' happens after XP is awarded for example), but the primary difference is the event source: 'KillMail' has the killer as the event source, whereas 'UnitDied' has the unit that died as the event source. This ability only wants to proc for the killer, not the unit that died. By using 'KillMail', you can use `eFilter_Unit` and a standard ability trigger function, simplifying the code a fair bit.

File: LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Ability_SkirmisherAbilitySet_LW.uc [hunk 6]
```diff
@@ -359,3 +368,157 @@ static function X2AbilityTemplate Interrupt_LW()
...
+	Template.eAbilityIconBehaviorHUD = eAbilityIconBehavior_NeverShow;
+	Template.Hostility = eHostility_Neutral;
+
+	Template.AbilityToHitCalc = default.DeadEye;
+	Template.AbilityTargetStyle = default.SelfTarget;
+	
+	EventListenerTrigger = new class'X2AbilityTrigger_EventListener';
+	EventListenerTrigger.ListenerData.Deferral = ELD_OnStateSubmitted;
+	EventListenerTrigger.ListenerData.EventID = 'BattlefieldAwarenessTriggered';
+	EventListenerTrigger.ListenerData.Filter = eFilter_None;
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Ability_SkirmisherAbilitySet_LW.uc:483]: Why not use `eFilter_Unit` with a standard ability trigger event function?

**Review by pledbrook — APPROVED** [review]
Good work, thanks.

**Pull request merged**